posthog/clickhouse/migrations/0037_kafka_engine_tables_dlq.py

```diff
@@ -0,0 +1,92 @@
+from infi.clickhouse_orm import migrations
+
+from posthog.clickhouse.plugin_log_entries import (
+    KAFKA_PLUGIN_LOG_ENTRIES_DLQ_SQL,
+    KAFKA_PLUGIN_LOG_ENTRIES_TABLE_SQL,
+    PLUGIN_LOG_ENTRIES_TABLE_MV_SQL,
+)
+from posthog.models.app_metrics.sql import (
+    APP_METRICS_MV_TABLE_SQL,
+    KAFKA_APP_METRICS_DLQ_SQL,
+    KAFKA_APP_METRICS_TABLE_SQL,
+)
+from posthog.models.event.sql import (
+    EVENTS_TABLE_JSON_MV_SQL,
+    KAFKA_EVENTS_TABLE_JSON_DLQ_SQL,
+    KAFKA_EVENTS_TABLE_JSON_SQL,
+)
+from posthog.models.group.sql import GROUPS_TABLE_MV_SQL, KAFKA_GROUPS_TABLE_DLQ_SQL, KAFKA_GROUPS_TABLE_SQL
+from posthog.models.ingestion_warnings.sql import (
+    INGESTION_WARNINGS_MV_TABLE_SQL,
+    KAFKA_INGESTION_WARNINGS_DLQ_SQL,
+    KAFKA_INGESTION_WARNINGS_TABLE_SQL,
+)
+from posthog.models.person.sql import (
+    KAFKA_PERSON_DISTINCT_ID2_DLQ_SQL,
+    KAFKA_PERSON_DISTINCT_ID2_TABLE_SQL,
+    KAFKA_PERSONS_DLQ_SQL,
+    KAFKA_PERSONS_TABLE_SQL,
+    PERSON_DISTINCT_ID2_MV_SQL,
+    PERSONS_TABLE_MV_SQL,
+)
+from posthog.models.session_recording_event.sql import (
+    KAFKA_SESSION_RECORDING_EVENTS_DLQ_SQL,
+    KAFKA_SESSION_RECORDING_EVENTS_TABLE_SQL,
+    SESSION_RECORDING_EVENTS_TABLE_MV_SQL,
+)
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+# Sets up a dead letter queue materialized view for each Kafka Engine table
+# This view stores the raw messages and errors we encountered when trying to insert
+# data into the main table via the Kafka Engine table + corresponding materialized view
+
+operations = [
+    # events_json
+    migrations.RunSQL(f"DROP TABLE events_json_mv ON CLUSTER '{CLICKHOUSE_CLUSTER}'"),
+    migrations.RunSQL(f"DROP TABLE kafka_events_json ON CLUSTER '{CLICKHOUSE_CLUSTER}'"),
+    migrations.RunSQL(KAFKA_EVENTS_TABLE_JSON_SQL()),
+    migrations.RunSQL(KAFKA_EVENTS_TABLE_JSON_DLQ_SQL()),
+    migrations.RunSQL(EVENTS_TABLE_JSON_MV_SQL()),
```

Comment:
This is going to cause issues in production due to the backup nodes - we should drop all the MV tables from them after this is live ASAP to avoid ingestion going down.